src/Mod/Part/App/Geometry.cpp

```diff
@@ -4884,12 +4884,19 @@ bool GeomSurface::normal(double u, double v, gp_Dir& dir) const
 
     Tools::getNormal(s, u, v, Precision::Confusion(), dir, done);
 
-    if (done)
+    if (done) {
         return true;
+    }
 
     return false;
 }
 
+std::optional<Base::Vector3d> GeomSurface::point(double u, double v) const
+{
+    Handle(Geom_Surface) s = Handle(Geom_Surface)::DownCast(handle());

```

Comment:
Can't auto be used here?

```suggestion
    auto s = Handle(Geom_Surface)::DownCast(handle());
```

Reply:
While I agree with it being better solution, it would be outlier in that file. What do you think?

Reply:
Let's keep it for now then - I want to go over that API later and change out arguments into returning optionals so the API is more idiomatic and then it would be a good time to fix that.